### PR TITLE
Simplify devcontainer setup

### PR DESCRIPTION
**Pull request by ludeeus**
The same setup as was added to https://github.com/home-assistant/addons-development

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,47 +1,25 @@
...
         libpulse0 \
-        git \
-        tar \
-        xz-utils \
-    && rm -rf /var/lib/apt/lists/* \
-    \
-    && curl -SL "https://storage.googleapis.com/shellcheck/shellcheck-v${SHELLCHECK_VERSION}.linux.x86_64.tar.xz" | tar xJf - \
-    && cp "shellcheck-v${SHELLCHECK_VERSION}/shellcheck" /usr/bin/ \
-    && rm -rf shellcheck-v${SHELLCHECK_VERSION}
+        shellcheck
```
Comment (pvizeli) [inline .devcontainer/Dockerfile:19]: How old is the shellcheck?

Comment (ludeeus) [reply .devcontainer/Dockerfile:19]: Not that bad, it's from march (0.5.0)

Comment (pvizeli) [reply .devcontainer/Dockerfile:19]: Do we not want to be close as possible to CI?

Comment (frenck) [reply .devcontainer/Dockerfile:19]: Alternatively, there are statically linked versions available on their github releases page...

<https://github.com/koalaman/shellcheck/releases>

Comment (ludeeus) [reply .devcontainer/Dockerfile:19]: Yeah, if not using the one in apt, we should use 'stable'

**Review by pvizeli — APPROVED** [review]

**Pull request merged**